### PR TITLE
Add support for saving documents using routing keys

### PR DESCRIPTION
**Pull request by dakota**
This adds support for saving documents with routing keys.

### CODE REVIEW

File: src/Index.php [hunk 1]
```diff
@@ -637,6 +641,7 @@ public function save(EntityInterface $entity, $options = [])
         $options += [
             'checkRules' => true,
             'refresh' => false,
+            'routing' => null,
```
Comment (markstory) [inline src/Index.php:644]: Could we add this to the docblock too?

File: src/Index.php [hunk 2]
```diff
@@ -555,6 +555,7 @@ public function saveMany($entities, $options = [])
...
```
Comment (markstory) [inline src/Index.php:558]: Could we add this to the docblock too?

Comment (dakota) [reply src/Index.php:559]: Done

File: src/Index.php (hunk 1 above)
Comment (dakota) [reply src/Index.php:646]: Done

File: src/Index.php [hunk 3]
```diff
@@ -666,6 +673,9 @@ public function save(EntityInterface $entity, $options = [])
 
         $doc = new ElasticaDocument($id, $data);
         $doc->setAutoPopulate(true);
+        if ($options['routing']) {
```
Comment (othercorey) [inline src/Index.php:676]: Can the routing be 0?

Comment (dakota) [reply src/Index.php:676]: I suppose that technically it can be, will change.

**Pull request merged**